### PR TITLE
add tpm pcr operation tests

### PR DESCRIPTION
**Pull request by PrzemyslawBanasiak13**
Signed-off-by: pbanasiak <przemyslaw.banasiak@3mdeb.com>

### CODE REVIEW

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md [hunk 1]
```diff
@@ -404,3 +404,374 @@ Example output:
 ```text
 SpecVersion=2.0
 ```
+
+## TPM003.001 PCRREAD Function (Ubuntu 22.04)
+
+**Test description**
+
+This test aims to verify that PCRREAD function works properly
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-security/200-tpm-support.md:412]: No dot at the end of the phrase

Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-security/200-tpm-support.md:412]: In my opinion, in the test description section, we should describe at least in a few words, what the function does.

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md [hunk 2]
```diff
@@ -404,3 +404,374 @@ Example output:
 ```text
 SpecVersion=2.0
 ```
+
+## TPM003.001 PCRREAD Function (Ubuntu 22.04)
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-security/200-tpm-support.md:408]: The title of the test case should be improved. In this test case we check the correctness of the function.

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md [hunk 3]
```diff
@@ -404,3 +404,374 @@ Example output:
...
+
+    ```bash
+    sudo tpm2_pcrread
+    ```
+
+1. Note the result
+
+**Expected result**
+
+1. Output from the command shoul look like below:
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-security/200-tpm-support.md:446]: `Should look like below`? I will ask in detail and in a clear way what are the conditions for passing this test.

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md [hunk 4]
```diff
@@ -404,3 +404,374 @@ Example output:
...
+        21: 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
+        22: 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
+        23: 0x0000000000000000000000000000000000000000000000000000000000000000
+    ```
+
+## TPM004.001 PCRALLOCATE Function (Ubuntu 22.04)
+
+**Test description**
+
+This test aims to verify that `PCRALLOCATE` function works properly.
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-security/200-tpm-support.md:505]: And what is that function responsible for?

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md [hunk 5]
```diff
@@ -404,3 +404,374 @@ Example output:
...
+
+    ```bash
+    sudo tpm2_pcrallocate
+    ```
+
+1. Note the result
+
+**Expected result**
+
+1. Output from the command shoul look like below:
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-security/200-tpm-support.md:539]: What that output contains? What are the conditions for passing the test?

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md [hunk 6]
```diff
@@ -404,3 +404,374 @@ Example output:
...
+
+1. Proceed with the
+    [Test cases common documentation](#test-cases-common-documentation) section.
+1. Install `tpm2-tools` using:
+
+    ```bash
+    sudo apt-get install tpm2-tools
+    ```
+
+    [Building from source](https://tpm2-tools.readthedocs.io/en/latest/INSTALL/)
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-security/200-tpm-support.md:569]: What exactly is this link responsible for?

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md [hunk 7]
```diff
@@ -404,3 +404,374 @@ Example output:
...
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-security/200-tpm-support.md:621]: What exactly is this link responsible for?

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md [hunk 8]
```diff
@@ -404,3 +404,374 @@ Example output:
...
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-security/200-tpm-support.md:675]: What exactly is this link responsible for?

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md [hunk 9]
```diff
@@ -404,3 +404,374 @@ Example output:
...
+
+    ```bash
+    sudo tpm2_pcrread
+    ```
+
+1. Note the results
+
+**Expected result**
+
+1. Output from **1st** `pcrread` should be similar:
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-security/200-tpm-support.md:710]: Are you sure this describes the output from the first command (tpm2_pcrextend)?

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md [hunk 10]
```diff
@@ -404,3 +404,374 @@ Example output:
...
+1. If process suceeds, last command should give similar output:
+
+    ```bash
+    sha1: f1d2d2f924e986ac86fdf7b36c94bcdf32beec15
+    sha256: b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c
+    sha384: 8effdabfe14416214a250f935505250bd991f106065d899db6e19bdc8bf648f3ac0f1935c4f65fe8f798289b1a0d1e06
+    sm3_256: 2b14a1fc49869413b0beb707069cffc0c6b0a51f3fedb9ce072c80709652b3ae
+    ```
+
+## TPM007.001 PCREXTEND And PCRRESET Functions (Ubuntu 22.04)
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-security/200-tpm-support.md:653]: Why were these two commands combined into one test case?

**Review by sulewskiprzemyslaw — CHANGES_REQUESTED** [review]

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md (hunk 10 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-security/200-tpm-support.md:653]: They're doing the opposite - PCREXTEND adds/modifies entries in PCR banks, nad PCRRESET resets/removes those entries

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md (hunk 1 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-security/200-tpm-support.md]: Fixed

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md (hunk 3 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-security/200-tpm-support.md]: Added explanations

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md (hunk 4 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-security/200-tpm-support.md]: Updated

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md (hunk 6 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-security/200-tpm-support.md]: Added context

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md (hunk 7 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-security/200-tpm-support.md]: Updated

File: docs/unified-test-documentation/dasharo-security/200-tpm-support.md (hunk 8 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-security/200-tpm-support.md]: Updated

**Review by sulewskiprzemyslaw — APPROVED** [review]

**Review by sulewskiprzemyslaw — APPROVED** [review]

**Pull request merged**